src/Nethermind/Nethermind.Optimism/OptimismPlugin.cs

```diff
@@ -0,0 +1,326 @@
+    private ManualBlockFinalizationManager? _blockFinalizationManager;
+    private IPeerRefresher? _peerRefresher;
+    private IBeaconPivot? _beaconPivot;
+    private BeaconSync? _beaconSync;
+    private IManualBlockProductionTrigger? _blockProductionTrigger;
+    private OptimismPostMergeBlockProducer? _blockProducer;
+
+    public bool ShouldRunSteps(INethermindApi api) => api.ChainSpec.SealEngineType == SealEngineType;
+
+    #region IConsensusPlugin
+
+    public string SealEngineType => Core.SealEngineType.Optimism;
+
+    public IBlockProductionTrigger DefaultBlockProductionTrigger => throw new NotImplementedException();
+
+    public Task<IBlockProducer> InitBlockProducer(IBlockProductionTrigger? blockProductionTrigger = null,
+        ITxSource? additionalTxSource = null)
+    {
+        if (blockProductionTrigger is not null || additionalTxSource is not null)
+            throw new ArgumentException(
+                "Optimism does not support custom block production trigger or additional tx source");
+
+        if (_api is null) throw new NullReferenceException(nameof(_api));
+
+        ArgumentNullException.ThrowIfNull(_api.SpecProvider);
+        ArgumentNullException.ThrowIfNull(_api.DbProvider);
+        ArgumentNullException.ThrowIfNull(_api.BlockTree);
+        ArgumentNullException.ThrowIfNull(_api.ReadOnlyTrieStore);
+        ArgumentNullException.ThrowIfNull(_api.BlockValidator);
+        ArgumentNullException.ThrowIfNull(_api.RewardCalculatorSource);
+        ArgumentNullException.ThrowIfNull(_api.ReceiptStorage);
+        ArgumentNullException.ThrowIfNull(_api.TxPool);
+        ArgumentNullException.ThrowIfNull(_api.TransactionComparerProvider);
+
+        _api.BlockProducerEnvFactory = new OptimismBlockProducerEnvFactory(
+            _api.ChainSpec,
+            _api.DbProvider,
+            _api.BlockTree,
+            _api.ReadOnlyTrieStore,
+            _api.SpecProvider,
+            _api.BlockValidator,
+            _api.RewardCalculatorSource,
+            _api.ReceiptStorage,
+            _api.BlockPreprocessor,
+            _api.TxPool,
+            _api.TransactionComparerProvider,
+            _blocksConfig,
+            _api.LogManager);
+
+        _api.GasLimitCalculator = new OptimismGasLimitCalculator();
+
+        _blockProductionTrigger = new BuildBlocksWhenRequested();
+
+        BlockProducerEnv producerEnv = _api.BlockProducerEnvFactory.Create();
+
+        _api.BlockProducer = _blockProducer = new OptimismPostMergeBlockProducer(
+            new OptimismPayloadTxSource(),
+            producerEnv.TxSource,
+            producerEnv.ChainProcessor,
+            producerEnv.BlockTree,
+            _blockProductionTrigger,
+            producerEnv.ReadOnlyStateProvider,
+            _api.GasLimitCalculator,
+            NullSealEngine.Instance,
+            new ManualTimestamper(),
+            _api.SpecProvider,
+            _api.LogManager,
+            _api.Config<IBlocksConfig>());
```

Comment:
Potentially could be done in `InitializeBlockProducer` step if we already need to overwrite steps